src/coreclr/src/vm/nativeimage.cpp

```diff
@@ -98,12 +101,17 @@ NativeImage::~NativeImage()
 NativeImage *NativeImage::Open(
     LPCWSTR fullPath,
     LPCUTF8 nativeImageFileName,
-    LoaderAllocator *pLoaderAllocator,
-    AllocMemTracker *pamTracker)
+    AssemblyLoadContext *pAssemblyLoadContext,
+    LoaderAllocator *pLoaderAllocator)
 {
     STANDARD_VM_CONTRACT;
 
     NewHolder<PEImageLayout> peLoadedImage = PEImageLayout::LoadNative(fullPath);
```

Comment:
Does this work on non-Windows? I do not think that `PEImageLayout::LoadNative` will keep returning the same address for multiple loads of the same image.

Reply:
Resolved in 2nd commit by changing the dictionary to be keyed by simple file names of the composite images. This is also related to DavidWr's feedback - I have reverted encoding the reverse relative paths in the OwnerCompositeExecutable R2R sections of the component assemblies and instead I added a new COMPlus variable <code>NativeImageSearchPaths</code> to let the runtime locate the native image in an arbitrary location (in addition to the containing folder of the MSIL assembly).